backend/git_real/permissions.py

```diff
@@ -22,6 +22,25 @@ def has_permission(self, request, view):
         correct = recieved_digest == "sha1=" + digest
         if not correct:
             return False
-            # breakpoint()
 
         return correct
+
+def user_can_see_repository(repo, user):
+    for project in repo.recruit_projects.all():
+        if user in project.recruit_users.all():
+            return True
+        if user in project.reviewer_users.all():
+            return True
+    return False
+
+class IsFilteredByRepoAttachedToProjectICanSee(BasePermission):
+
+    def has_permission(self, request, view):
+        if request.method != "GET":
```

Comment:
The name of this filter class doesn't imply that the request must be a GET. Why not just check that the user has permission to see the repo? You cal check the request method elsewhere. 

In fact you do check the request method elsewhere. If the ActionIs(list) then it's definately a GET request.